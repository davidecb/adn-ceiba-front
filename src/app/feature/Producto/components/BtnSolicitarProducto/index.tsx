import * as PropTypes from 'prop-types';
import * as React from 'react';
import { Button } from 'app/shared/components/Button';
import { Producto } from '../../models/Producto';

interface BtnSolicitarProductoProps {
  onSolicitar: (producto: Producto) => any;
  producto: Producto;
}

export const BtnSolicitarProducto: React.FC<BtnSolicitarProductoProps> = ({
  onSolicitar,
  producto,
}) => {
  const handleSolicitar = () => onSolicitar(producto);
  return (
    <Button onClick={handleSolicitar}>
      <span>
        Solicitar
      </span>
    </Button>
  );
};

BtnSolicitarProducto.propTypes = {
  producto: PropTypes.shape({
    id: PropTypes.number.isRequired,
    nombre: PropTypes.string.isRequired,
    costo: PropTypes.number.isRequired,
    tiempo: PropTypes.number.isRequired,
    imagen: PropTypes.string.isRequired
  }).isRequired,
  onSolicitar: PropTypes.func.isRequired,
};
