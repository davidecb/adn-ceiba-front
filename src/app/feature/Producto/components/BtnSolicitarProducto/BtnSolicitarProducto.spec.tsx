import * as React from 'react';
import { RenderResult, render } from '@testing-library/react';
import { SinonStub, stub } from 'sinon';
import { BtnSolicitarProducto } from './../BtnSolicitarProducto';

describe('BtnSolicitarProducto Test', () => {
  let componentWrapper: RenderResult;
  let componentProps: React.ComponentProps<typeof BtnSolicitarProducto> & {
    onSolicitar: SinonStub;
  };

  beforeEach(() => {
    componentProps = {
      producto: {
        id: 1,
        nombre: 'producto testing',
        costo: 10000,
        tiempo: 30,
        imagen: 'imagen.jpg'
      },
      onSolicitar: stub(),
    };
    componentWrapper = render(<BtnSolicitarProducto {...componentProps} />);
  });

  it('should match snapshot', () => {
    expect(componentWrapper.container).toMatchSnapshot();
  });
});

export {};
