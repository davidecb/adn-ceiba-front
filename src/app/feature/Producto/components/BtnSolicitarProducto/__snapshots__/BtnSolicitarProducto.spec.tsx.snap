// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BtnSolicitarProducto Test should match snapshot 1`] = `
<div>
  <button
    class="sc-AxjAm dmxFrE"
  >
    <span>
      Solicitar
    </span>
  </button>
</div>
`;
